sig-provider/sig-provider/src/aggregator.rs

```diff
@@ -129,6 +153,48 @@ fn decode_txinput(args: &[ParamType], tx_args: &[u8]) -> Option<Vec<Token>> {
     Some(decoded)
 }
 
+fn decode_log(name: String, args: &[ParamType], raw: RawLog) -> Option<(Vec<Token>, Vec<bool>)> {
+    const MAX_PERMUTATIONS: usize = 10000;
+    // this is indeed can be very long
+    // there are better ways to iterate over valid indexed permutations
+    // but right now we think this is okayish
+    for (ind, perm) in (0..raw.topics.len() - 1)
```

Comment:
Could you please add a description how that algorithm works and why it does it that way? So that it will be at least clear why such a complex actions are required in the first place